yt/java/ytsaurus-client/src/main/java/tech/ytsaurus/client/operations/MapperOrReducerSpec.java

```diff
@@ -95,6 +90,42 @@ protected MapperOrReducerSpec(Class<?> mainClazz, Builder<?> builder) {
         prepareTimeLimit = builder.prepareTimeLimit;
     }
 
+    // Copied from `BaseLayerDetector.guess_base_layers`
+    public static StartOperation guessBaseLayers(StartOperation req) {
+        var userLayer = System.getenv("YT_BASE_LAYER");
+        if (userLayer == null) {
+            return req;
+        }
+        userLayer = userLayer.toLowerCase().strip();
+        var spec = req.getSpec();
+        if (spec.getAttribute("layer_paths").isPresent() || spec.getAttribute("docker_image").isPresent()) {
+            // do not change user spec
+            if (Arrays.asList("auto", "porto:auto", "docker:auto").contains(userLayer)) {
+                logger.debug("Operation has layer spec. Do not guess base layer");
+            }
+            return req;
+        }
+        if (Arrays.asList("auto", "docker:auto", "porto:auto").contains(userLayer)) {
+            // TODO: implement later
+            return req;
```

Comment:
Wouldn't it be better to throw exception here?

Reply:
To break less, I would rather keep the current behavior for those cases: doing nothing.